app/assets/javascripts/float-kit/addon/components/d-menu.gjs

```diff
@@ -87,7 +92,7 @@ export default class DMenu extends Component {
   }
 
   <template>
-    <DButton
+    <this.triggerTag
```

Comment:
Bit weird that this component is going to have things like `@icon`, `@translatedAriaLabel` etc. passed into it...

How about making the argument `@triggerComponent`, and then the default can be something like

```
get triggerComponent(){
  return this.args.triggerComponent || curryComponent(DButton, { icon: , translatedLabel, ... });
}
```

Reply:
yes I agree with you that was bothering me, your solution sounds nice, will try to implement it